homeassistant/components/sensor/__init__.py

```diff
@@ -407,16 +445,48 @@ def state(self) -> Any:
                     f"but does not provide a date state but {type(value)}"
                 ) from err
 
+        if (
+            value is not None
+            and self.device_class in UNIT_CONVERSIONS
+            and native_unit_of_measurement != unit_of_measurement
```

Comment:
It's better to do the unit != check first, it's faster and most of the times it won't be different.
```suggestion
            and native_unit_of_measurement != unit_of_measurement
            and self.device_class in UNIT_CONVERSIONS
```